Userland/Libraries/LibDisassembly/riscv64/IM.cpp

```diff
@@ -0,0 +1,268 @@
+            operation = ArithmeticInstruction::Operation::Remainder;
+            break;
+        case 0b111:
+            operation = ArithmeticInstruction::Operation::RemainderUnsigned;
+            break;
+        default:
+            VERIFY_NOT_REACHED();
+        }
+    }
+    return adopt_own(*new ArithmeticInstruction(operation, raw_parts.rs1, raw_parts.rs2, raw_parts.rd));
+}
+
+NonnullOwnPtr<ArithmeticInstruction> parse_op_32(u32 instruction)
+{
+    auto raw_parts = RawRType::parse(instruction);
+    auto is_m_extension = (raw_parts.funct7 & 1) == 1;
+
+    auto operation = ArithmeticInstruction::Operation::Add;
+    if (!is_m_extension) {
+        switch (raw_parts.funct3) {
+        case 0b000:
+            if ((raw_parts.funct7 & 0b0100000) == 0)
+                operation = ArithmeticInstruction::Operation::AddWord;
+            else
+                operation = ArithmeticInstruction::Operation::SubtractWord;
+            break;
+        case 0b001:
+            operation = ArithmeticInstruction::Operation::ShiftLeftLogicalWord;
+            break;
+        case 0b101:
+            if ((raw_parts.funct7 & 0b0100000) == 0)
+                operation = ArithmeticInstruction::Operation::ShiftRightLogicalWord;
+            else
+                operation = ArithmeticInstruction::Operation::ShiftRightArithmeticWord;
+            break;
+        default:
+            VERIFY_NOT_REACHED();
+        }
+    } else {
+        switch (raw_parts.funct3) {
+        case 0b000:
+            operation = ArithmeticInstruction::Operation::MultiplyWord;
+            break;
+        case 0b100:
+            operation = ArithmeticInstruction::Operation::DivideWord;
+            break;
+        case 0b101:
+            operation = ArithmeticInstruction::Operation::DivideUnsignedWord;
+            break;
+        case 0b110:
+            operation = ArithmeticInstruction::Operation::RemainderWord;
+            break;
+        case 0b111:
+            operation = ArithmeticInstruction::Operation::RemainderUnsignedWord;
+            break;
+        default:
+            VERIFY_NOT_REACHED();
+        }
+    }
+    return adopt_own(*new ArithmeticInstruction(operation, raw_parts.rs1, raw_parts.rs2, raw_parts.rd));
+}
+
+NonnullOwnPtr<MemoryLoad> parse_load(u32 instruction)
+{
+    auto raw_parts = RawIType::parse(instruction);
+    auto width = MemoryAccessMode::from_funct3(raw_parts.funct3);
+    return adopt_own(*new (nothrow) MemoryLoad(raw_parts.imm, raw_parts.rs1, width, raw_parts.rd));
+}
+
+NonnullOwnPtr<MemoryStore> parse_store(u32 instruction)
+{
+    auto raw_parts = RawSType::parse(instruction);
+    auto width = MemoryAccessMode::from_funct3(raw_parts.funct3);
+    return adopt_own(*new (nothrow) MemoryStore(raw_parts.imm, raw_parts.rs2, raw_parts.rs1, width));
+}
+
+NonnullOwnPtr<Branch> parse_branch(u32 instruction)
+{
+    auto raw_parts = RawBType::parse(instruction);
+    auto condition = static_cast<Branch::Condition>(raw_parts.funct3);
```

Comment:
If funct3 is 0b010 or 0b011, `condition` has an invalid value, but you don't catch invalid enum values in Formatting.cpp.